text/3691-trusted-publishing-cratesio.md

```diff
@@ -0,0 +1,371 @@
+
+For GitHub Actions, the crate author will need to provide:
+
+1. (required) The owning GitHub username or organization
+2. (required) The repository name
+3. (required) The workflow file name (must be located in `.github/workflows/`)
+4. (optional) The [GitHub Actions environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) name
+
+
+### Example GitHub Actions Workflow
+[example-gha-workflow]: #example-gha-workflow
+
+With a _Trusted Publisher Configuration_ in place, a example GitHub Actions workflow may look like this:
+
+```yaml
+name: Publish Crate
+
+# Some may opt to use a trigger based on a push with a tag instead
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    # here we use a GitHub Actions _Environment_ that allows for more 
+    # control and steps to ensure security of the publishing 
+    # process
+    # 
+    # See https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment
+    environment: release
+
+    permissions:
+      id-token: write  # required in order to get a signed ID token
+      contents: read
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install dependencies
+      run: cargo build --release
+
+    # This action does not exist currently.
+    - name: Authenticate with crates.io
+      id: auth
+      uses: rust-lang/crates-io-auth-action@v1
+
+    - name: Publish to crates.io
+      run: cargo publish
+      env:
+        CARGO_REGISTRY_TOKEN: ${{ steps.auth.outputs.token }}
+
+```
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Required Configuration Parameters
+[configuration]: #configuration
+
+As stated in the [Trusted Publisher Configuration on crates.io](#trusted-publisher-configuration-cratesio) section, a configuration will need to provide:
+
+1. (required) The owning GitHub username or organization
+2. (required) The repository name
+3. (required) The workflow file name (must be located in `.github/workflows/`)
+4. (optional) The [GitHub Actions environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment) name
+
+The crate owner will need to provide which crate the particular Trusted Publisher Configuration is allowed to publish. 
+
+The Trusted Publisher Configuration might look like the below, represented in JSON format:
+
+```json
+{
+    "name": "trusted-publisher-config-for-sampleproject",
+    "required-claims": {
+        "owner": "octo-org",
+        "repo": "sampleproject",
+        "workflow": "release.yml",
+        "environment": "release"
+    },
+    "crate": "my-sample"
+}
+```
+
+In addition to the above parameters, the configuration will be linked to a particular crates.io user or team to administer.
```

Comment:
Agreed, it makes more sense to make the relationship between the project and the publisher, and confer ability to modify the configuration on any user who can otherwise modify the project.

Reply:
You're correct. I worded this poorly.
I'll update this. :)